src/stream/src/executor/top_n/group_top_n.rs

```diff
@@ -246,34 +247,35 @@ impl<const WITH_TIES: bool> GroupTopNCache<WITH_TIES> {
     }
 }
 #[async_trait]
-impl<S: StateStore, const WITH_TIES: bool> TopNExecutorBase
-    for InnerGroupTopNExecutorNew<S, WITH_TIES>
+impl<K: HashKey, S: StateStore, const WITH_TIES: bool> TopNExecutorBase
+    for InnerGroupTopNExecutorNew<K, S, WITH_TIES>
 where
     TopNCache<WITH_TIES>: TopNCacheTrait,
 {
     async fn apply_chunk(&mut self, chunk: StreamChunk) -> StreamExecutorResult<StreamChunk> {
         let mut res_ops = Vec::with_capacity(self.limit);
         let mut res_rows = Vec::with_capacity(self.limit);
+        let chunk = chunk.compact();
```

Comment:
Is the compaction necessary? 🤔

Reply:
Yes, if not, we can not ensure the` keys.size()` equal to `chunk.rows()`.

Reply:
`HashKey::build` will not skip invisible row.